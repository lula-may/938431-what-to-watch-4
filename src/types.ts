export interface Review {
  author: string;
  authorId: number;
  date: Date;
  id: number;
  rating: number;
  text: string;
}

export interface Movie {
  actors: string[];
  bgColor: string;
  bgPoster: string;
  src: string;
  description: string;
  director: string;
  genre: string;
  id: number;
  isFavorite: boolean;
  poster: string;
  previewPoster: string;
  rating: {
    count: number,
    score: number,
  };
  releaseYear: number;
  runTime: number;
  title: string;
};
