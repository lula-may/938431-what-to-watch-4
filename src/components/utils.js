import {DEFAULT_GENRE} from "../const.js";

const getMoviesByGenre = (genre, allMovies) => {
  if (genre === DEFAULT_GENRE) {
    return allMovies;
  }
  return allMovies.filter((movie) => movie.genre === genre);
};

export {getMoviesByGenre};
