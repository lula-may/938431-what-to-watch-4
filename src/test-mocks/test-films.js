const movieTitles = [
  `Gone With The Wind`,
  `The Godfather`,
  `Casablanca`,
  `A Streetcar Named Desire`,
  `Some Likes It Hot`,
];

const movies = movieTitles.map((title, i) => {
  return {
    title,
    id: i,
    src: `poster-${i}.jpg`
  };
});

const headerMovie = {
  title: `Jurassic Park`,
  genre: `Sci-Fi`,
  releaseYear: 1993,
  bg: `background.jpg`,
  poster: `header-poster.jpg`
};

export {movies, headerMovie};

