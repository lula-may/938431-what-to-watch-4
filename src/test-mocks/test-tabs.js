const tabs = [
  {
    id: `aa`,
    type: `First`
  },
  {
    id: `bb`,
    type: `Second`,
  },
  {
    id: `cc`,
    type: `Third`,
  }
];

export {tabs};
