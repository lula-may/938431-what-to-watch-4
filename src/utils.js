const HOUR = 60;
const DATE_STRING_LENGTH = 10;
const SECONDS_IN_HOUR = 3600;
const SECONDS_IN_MINUTE = 60;

const extend = (a, b) => Object.assign({}, a, b);

const formatTime = (duration) => {
  const hours = Math.floor(duration / HOUR);
  const minutes = duration % HOUR;

  return (
    (hours > 0)
      ? `${hours}h ${minutes}m`
      : `${minutes}m`
  );
};

const getTwoDigitNumber = (number) => number < 10 && `0${number}` || `${number}`;

const formatElapsedTime = (durationInSeconds) => {
  const duration = Math.round(durationInSeconds);
  const hours = Math.floor(duration / SECONDS_IN_HOUR);
  const minutes = Math.floor(duration % SECONDS_IN_HOUR / SECONDS_IN_MINUTE);
  const seconds = duration % SECONDS_IN_MINUTE;
  return `${getTwoDigitNumber(hours)}:${getTwoDigitNumber(minutes)}:${getTwoDigitNumber(seconds)}`;
};

const formatDate = (date) => date.toLocaleString(`en-US`, {day: `numeric`, month: `long`, year: `numeric`});

const formatDateTimeAttribute = (date) => date.toISOString().slice(0, DATE_STRING_LENGTH);

export {extend, formatDate, formatTime, formatDateTimeAttribute, formatElapsedTime};
