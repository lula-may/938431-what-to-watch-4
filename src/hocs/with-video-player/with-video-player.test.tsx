import * as React from "react";
import * as renderer from "react-test-renderer";

import withVideoPlayer from "./with-video-player";
import {testMovies} from "../../test-mocks/test-films";
import {Movie} from "../../types";
import {noop} from "../../utils";

const movie: Movie = testMovies[0];

interface Props {
  children: React.ReactNode;
  onCardClick: () => void;
  onCardEnter: () => void;
}

const MockComponent: React.FC<Props> = (props: Props) => {
  const {children, onCardClick, onCardEnter} = props;
  return (
    <div onMouseEnter={onCardEnter} onClick={onCardClick}>
      {children}
    </div>
  );
};


const MockComponentWrapped = withVideoPlayer(MockComponent);

describe(`WithVideoPlayer HOC`, () => {
  it(`should render <video/> inside MockComponent`, () => {
    const tree = renderer.create(
        <MockComponentWrapped
          movie={movie}
          onCardEnter={noop}
          onCardClick={noop}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});


