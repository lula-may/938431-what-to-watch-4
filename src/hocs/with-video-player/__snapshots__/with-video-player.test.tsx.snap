// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithVideoPlayer HOC should render <video/> inside MockComponent 1`] = `
<div
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <video
    height="175"
    poster="poster-0.jpg"
    width="280"
  />
</div>
`;
